CustomCollections/CustomCollections/CustomVector.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomVector<T> : IList<T>
+    {
+        private T[] _vector;
+
+        public int Count { get; private set; } = 0;
+        public int Length { get; private set; } = 1;
+        public bool IsReadOnly { get; } = false;
+
+        public CustomVector() => _vector = new T[1];
+
+        public T this[int index]
+        {
+            get => _vector[index];
+            set => _vector[index] = value;
+        }
+
+        public void Add(T item)
+        {
+            if(!IsEmptyPosition()) GrowSize();
+            _vector[Count] = item;
+            Count++;
+        }
+
+        public void Clear()
+        {
+            _vector = new T[1];
+            Count = 0;
+            Length = 1;
+        }
+
+        public bool Contains(T item)
```

Comment:
Notice that this method is exactly the same as IndexOf, except this one returns true/false instead of index. You can simply reuse IndexOf here.